yt/loaders.py

```diff
@@ -1457,3 +1463,108 @@ def load_sample(
         loadable_path = loadable_path.joinpath(load_name, specific_file)
 
     return load(loadable_path, **kwargs)
+
+
+# --- Loader for tar-based datasets ---
+def load_archive(
+    fn: Union[str, Path],
+    path: str,
+    ratarmount_kwa: Optional[Dict] = None,
+    *args,
+    **kwargs,
+) -> Dataset:
+    r"""
+    Load archived data with yt.
+
+    This is a wrapper around :func:`~yt.loaders.load` to include mounting
+    and unmounting the archive as a read-only filesystem and load it.
+
+    Parameters
+    ----------
+
+    fn: str
+        The `filename` of the archive containing the dataset.
+
+    path: str
+        The path to the dataset in the archive.
+
+    ratarmount_kwa: dict, optional
+        Optional parameters to pass to ratarmount to mount the archive.
+
+    Notes
+    -----
+
+    - Any additional keyword argument is passed down to :func:`~yt.loaders.load`.
+    - This function requires ratarmount.
+    - This function does not work on Windows system.
+    """
+    if platform.system() not in ("Linux", "Darwin"):
+        raise RuntimeError(
+            "Reading from an archive is not supported on your operating system"
+            f" ({platform.system()})."
+        )
+
+    if isinstance(fn, Path):
+        fn = str(fn.resolve().absolute())
+
+    if ratarmount_kwa is None:
+        ratarmount_kwa = {}
+
+    with open(fn, mode="br") as fd:
+        compression = ratarmount.SQLiteIndexedTar._detectCompression(fd)
+
+    if compression is None:
+        raise YTUnidentifiedDataType(fn, *args, **kwargs)
+
+    # Note: the temporary directory will be created by ratarmount
+    tempdir = fn + ".mount"
+    tempdir_base = tempdir
+    i = 0
+    while os.path.exists(tempdir):
+        i += 1
+        tempdir = f"{tempdir_base}.{i}"
+
+    def mount(archive: str, mountPoint: str, ratarmount_kwa: Dict):
+        fuseOperationsObject = ratarmount.TarMount(
+            pathToMount=archive,
+            mountPoint=mountPoint,
+            lazyMounting=True,
+            **ratarmount_kwa,
+        )
+        ratarmount.fuse.FUSE(
+            operations=fuseOperationsObject,
+            mountpoint=mountPoint,
+            foreground=True,
+            nothreads=True,
+        )
+
+    proc = Process(target=mount, args=(fn, tempdir, ratarmount_kwa))
+    proc.start()
+    # Note: the mounting needs to happen in another process which
+    # needs be run in the foreground (otherwise it may
+    # unmount). To prevent a race-condition here, we wait
+    # for the folder to be mounted within a reasonable time.
+    retry = 0
+    while retry < 100:
```

Comment:
up to 10s of looping in vain seems a bit excessive to me. 10 attempt should be more than enough, shouldn't it ?